lib/cli.rb

```diff
@@ -0,0 +1,40 @@
+require_relative "recipe_repository"
+require_relative "local_data_source"
+require_relative "api_client_with_syncing"
+require_relative "api_client"
+require_relative "plain_text_presenter"
+
+class Cli
+  DATA_FILE_PATH = "data/recipe_data.json"
+
+  def sample
+    recipe = recipe_repository.all_recipes.sample
```

Comment:
It's interesting to think about if the API allowed us to fetch a random recipe (i.e. we wouldn't have to fetch all, then sample), then it could read:

```ruby
recipe = recipe_repository.sample
```

However, that makes `sample` a core part of `recipe_repository`'s domain.

However, compare to Rails:
`Recipe.all.where(name: "foo").where(serves: 2)`
where we build up an intention that we eventually execute.
Could be worth thinking about similar things here.

Reply:
For example, requesting top 10 recipes by rating:
Could be either
```ruby
recipe_repository.top_10
```
and have recipe_repository implement that. But then business logic starts to creep into something which is potentially more generic.

Alternative:
```
recipe_repository.all.where("rating > 7").limit(1)
```
stealing some ideas from Rails.